src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -229,6 +288,42 @@ public Set<String> getAllPublishedVersionBranchPaths() {
 		return publishedVersionBranchPaths;
 	}
 	
+	public Set<String> getCurrentRefsetBranchPaths() {
+		List<Branch> allBranches = branchService.findAll();
+		Set<String> refsetBranchPaths = new HashSet<>();
+		Map<String,String> refsetToCurrentBranchPathMap = new HashMap<>();
+		//Identify the latest Refset edit branch for each refset
+		//For example:
+		//MAIN/SNOMEDCT/REFSETS/REFSET-48353439008-1676579557135/EDIT-1676579564596
+		//Is for refset 4853439008, timestamp 1676579564596
+		final Pattern p = Pattern.compile("^(.*)(REFSET-)([0-9]+)(-[0-9]+\\/EDIT-)([0-9]+)$");
```

Comment:
This is extremely client-application specific to be putting into a terminology server.

Reply:
You're not wrong.  But since this functionality is only accessible via the "refsetsOnly" parameter, having them reference the Refset tool specific branch patterns hopefully isn't too horrible.

I'm trying to think of other alternatives.  The problem we're trying to solve is that the existing MultiSearchService.findDescriptions functionality only searches published branches, and one of the Refset Tool requirements is to be able to search for refsets by description across all branches, including unpublished ones.  An alternative to hard coding the regex could be to expose something like "include branches" parameter in the api, and the refset tool could grab all branches, parse them, and pass the ones it wants in. But each branch is a long string and we would need to pass in _every_ in-progress refset branch, which would make a monster call...

Reply:
Sounds like you've some options there.  Like, I liked the ecl parameter because I said "oh that's something that people might be generally useful", but this code is so specific to the internal workings of your application that it can't be used anywhere else.    Have you time for a call in the next couple of hours @RichardAWood ?

Reply:
Hi Peter,
I'm online now, and have a call in 20 minutes (8:00-8:30 am PST) - want to try to sneak in a talk before or after that?